src/common/fast-path.js

```diff
@@ -111,6 +142,9 @@ FastPath.prototype.each = function each(callback /*, name1, name2, ... */) {
 // Similar to FastPath.prototype.each, except that the results of the
 // callback function invocations are stored in an array and returned at
 // the end of the iteration.
+/**
+ * @param {(path: FastPath, index: number) => any} callback
```

Comment:
I'm currently learning JSDoc and don't know how to type this in the following way:

```ts
map<U>(callback: (path: this, index: number) => U, ...names: PropertyKey[]): U[];
// ~~~                  ~~~~                    ~                            ~
```

Reply:
I think there’s an `@template U` or `@template {U}` tag.

Reply:
Yes, but I don't know how to apply it here
I have tested some combinations